data/binding/printf.go

```diff
@@ -0,0 +1,250 @@
+package binding
+
+import (
+	"fmt"
+)
+
+type sprintfString struct {
+	String
+
+	format string
+	source []DataItem
+	err    error
+}
+
+// NewSprintf returns a String binding that format its content using the
+// format string and the provide additional parameter that must be other
+// data bindings.
+//
+// Since: 2.2
+func NewSprintf(format string, b ...DataItem) (String, error) {
+	ret := &sprintfString{String: NewString(), format: format}
+
+	for _, value := range b {
+		value.AddListener(ret)
+
+		ret.source = append(ret.source, value)
+	}
+
+	return ret, nil
+}
+
+func (s *sprintfString) DataChanged() {
+	data := make([]interface{}, 0)
+
+	s.err = nil
+	for _, value := range s.source {
+		switch x := value.(type) {
+		case Bool:
+			{
+				b, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, b)
+				break
+			}
+		case Bytes:
+			{
+				b, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, b)
+				break
+			}
+		case Float:
+			{
+				f, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, f)
+				break
```

Comment:
Go adds break statements to switch cases by default (applies to more places).

Reply:
Thanks, I forgot!